sandbox/controller.go

```diff
@@ -68,6 +69,14 @@ func WithNetNSPath(netNSPath string) CreateOpt {
 	}
 }
 
+// WithAnnotations sets the provided annotations for sandbox creation.
+func WithAnnotations(annotations map[string]string) CreateOpt {
```

Comment:
I don't think we have any use for it in CRI given we send over the pod spec (although this is nice if we ever stopped doing that). This was more a general addition

Reply:
nod .. will give us the option of annotating the sandbox instead of the podsandbox in cases where the annotation is not wanted/needed to be published via CRI.

Reply:
Right, I think this could let us replace the pod sandbox config we're sending over today with a potentially more useful struct, and we can send the annotations as is in this field now. Win win 😈